### ISSUE TITLE
[NV-1924] notification failing due to absence of token in case of multiple active push providers

### ISSUE THREAD

**Issue by jainpawan21** [novuhq/novu#3214]
Steps to reproduce:- 

\- FCM -> active 

\- APNS -> active 

\- Subscriber has only FCM token set in credentials. 

\- Send push notification 

\- 2 actions have been triggered: 

\- sent with FCM successfully 

\- failed to send with APNS (due to lack of token)

Solution:- There can be option in override or in separate field to activate/deactivate specific provider during trigger

<sub>From [SyncLinear.com](https://synclinear.com) | [NV-1924](https://linear.app/novu/issue/NV-1924/notification-failing-due-to-absence-of-token-in-case-of-multiple)</sub>

**Comment by jainpawan21** [comment]
can we please add public label also so that may be someone from community can help us?

**Comment by jainpawan21** [comment]
help users see which integratiosn is active / dormant and which integration exist - this is the channel where we are asked for multiple integrations.

**Comment by jainpawan21** [comment]
We should allow for better management of active push integration IMO.

@nikolay  @scopsy  WDYT?

**Comment by SerafeimOikonomou** [comment]
I'll try to provide my feedback regarding this in case it helps.

- When there are 2 active providers (FCM and APNS) and the subscriber has both FCM and APNS tokens, Novu tries to send both notifications via FCM, which leads to an error because the iOS token is APNS and not FCM.
- When there are 2 active providers (FCM and APNS) and the subscriber has only an FCM token, then again Novu tries to send the notification twice via FCM but it fails again because the APNS token doesn't exist.

One problem is that, as it seems, it gets the first available provider and tries to send everything using that one. Since FCM can be used for both android and iOS devices I guess that a provider could be selected during triggering a notification so that the user is able to choose the correct provider that the token is associated with. However, I don't think that enabling/disabling providers during trigger could be a solution because it will create issues when sending multiple notifications in parallel.

**Comment by TNAJanssen** [comment]
See #3258 as it can be related

**Comment by scopsy** [comment]
@TNAJanssen yep, @jainpawan21 take a look at this in case we can connect the two issues and close this one

**Comment by TNAJanssen** [comment]
@scopsy this pull request probably solves the issue:

https://github.com/novuhq/novu/pull/3259
